financeextras.php

```diff
@@ -151,3 +151,13 @@ function financeextras_civicrm_links($op, $objectName, $objectId, &$links, &$mas
     $hook->run();
   }
 }
+
+/**
+ * Implements hook_civicrm_buildForm().
+ */
+function financeextras_civicrm_buildForm($formName, &$form) {
```

Comment:
Can we move the logic to check inside the class? Since the function will be growing quite big, and you can encapsulate if we should run adding button in the class, see how this was done in the link hook.  

https://github.com/compucorp/io.compuco.financeextras/blob/CIVIPLUS-768-workstream/Civi/Financeextras/Hook/Links/Contribution.php#L48 and Stripe extension https://github.com/compucorp/uk.co.compucorp.stripe/blob/master/stripe.php#L199

Reply:
@kuldip-compuco I think you missed this comment.